### PR TITLE
#1 Mockup of the antibodies table/1

### PR DESCRIPTION
**Pull request by Anto8819**
### Description

First draft of the Antibodies table in the home page
- Based on the DataGrid component from [MUI/x](https://mui.com/x/api/data-grid/data-grid/)
- Hardcoded JSON data within the `src/services `directory (real data from the Antibodies Registry current app)

### How to test this PR manually?

- In the frontend directory, run `npm i `and `npm start `to run the development server.
- In the browser, open http://localhost:9000

#### Expected result
Screen Shot 2022-07-29 at 13 06 42

### CODE REVIEW

File: applications/areg-portal/frontend/src/utils/helpers.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import dataJson from "./data.json";
```
Comment (filippomc) [inline applications/areg-portal/frontend/src/utils/helpers.ts:1]: As a convention, we use "utils" for static functions, "helpers" for control logic helpers (we don't usually have this on react as everything is on components or redux), "service" for function or objects that fetch data

**Review by filippomc — APPROVED** [review]
Great job @Anto8819! Dropped a few comments mainly on code conventions

File: applications/areg-portal/frontend/src/components/AntibodiesTable.tsx [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+import { Typography, Box } from "@mui/material";
+
+//project imports
+import { getAntibodies } from "../utils/helpers";
+import Content from "./UI/Content";
+
+const renderNameAndId = (props: GridRenderCellParams<String>) => {
+  return (
+    <Box>
+      <Typography variant="body2">{props.row.ab_name}</Typography>
```
[the remark this answers is not in the document]
Comment (Anto8819) [reply applications/areg-portal/frontend/src/components/AntibodiesTable.tsx:20]: Done

File: applications/areg-portal/frontend/src/App.tsx [hunk 3]
```diff
@@ -1,16 +1,17 @@
...
-    <>
-      <img src="/assets/icon.png" />
-      <h1>Sample React application is working!</h1>
-      <RestTest />
-      <p>See api documentation <a href="/api/ui">here</a></p>
-    </>
+  <BrowserRouter>
+    <Switch>
+      <Route exact path="/" component={LandingTest} />
+      <Route exact path="/list" component={AntibodiesTable} />
```
[the remark this answers is not in the document]
Comment (Anto8819) [reply applications/areg-portal/frontend/src/App.tsx:12]: Done

File: applications/areg-portal/frontend/src/components/AntibodiesTable.tsx [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+  );
+};
+
+const rows: GridRowsProp = getAntibodies();
+
+const columns: GridColDef[] = [
+  {
+    field: "ab_name",
+    headerName: "Name",
+    minWidth: 75,
```
[the remark this answers is not in the document]
Comment (Anto8819) [reply applications/areg-portal/frontend/src/components/AntibodiesTable.tsx:44]: Done

File: applications/areg-portal/frontend/src/components/AntibodiesTable.tsx [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+    <Box>
+      <Typography variant="body2">{props.row.ab_name}</Typography>
+      <Typography variant="caption" align="left" component="div">
+        {props.row.ab_id}
+      </Typography>
+    </Box>
+  );
+};
+
+const renderTargetAntigenAndSpecies = (props: GridRenderCellParams<String>) => {
```
[the remark this answers is not in the document]
Comment (Anto8819) [reply applications/areg-portal/frontend/src/components/AntibodiesTable.tsx:28]: you're right, my bad :)

File: applications/areg-portal/frontend/src/components/UI/Content.tsx [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import React from "react";
+import Container from "@mui/material/Container";
+import CssBaseline from "@mui/material/CssBaseline";
+
+const Content = ({ children }) => {
```
[the remark this answers is not in the document]
Comment (Anto8819) [reply applications/areg-portal/frontend/src/components/UI/Content.tsx:5]: done

File: applications/areg-portal/frontend/src/utils/helpers.ts (hunk 1 above)
Comment (Anto8819) [reply applications/areg-portal/frontend/src/utils/helpers.ts]: done

**Pull request merged**